ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -748,6 +748,11 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent keyEvent) {
                   new ReactTextInputSubmitEditingEvent(
                       editText.getId(),
                       editText.getText().toString()));
+              // if blurOnSubmit is false, and isMultiline, go to new line
```

Comment:
I would suggest rewrite whole function in this manner with local boolean variables: `blurOnSubmit`, `isMultiline`, `shouldDispatchSubmitEvent`, `shouldClearFocus`, and `returnValue`.